core-bundle/src/Routing/ResponseContext/CoreResponseContextFactory.php

```diff
@@ -94,7 +94,7 @@ public function createContaoWebpageResponseContext(PageModel $pageModel): Respon
                     throw new \RuntimeException('The request stack did not contain a request');
                 }
 
-                $url = UrlUtil::makeAbsolute($url, $request->getUri());
+                $url = UrlUtil::makeAbsolute($url, $request->getSchemeAndHttpHost().$request->getBasePath().'/');
```

Comment:
This is incorrect, we have to use the `basePath`, not the `baseUrl`

Reply:
```
public function getUriForPath(string $path): string
{
    return $this->getSchemeAndHttpHost().$this->getBaseUrl().$path;
}
```

Reply:
I don't think that's true. We don't want `getUri` because that contains the current path. But `getUriForPath` does not, and since these are page links we do want the entry point script (e.g. `preview.php`) to be in the links.

Reply:
The entry point is already included in the provided `$url` so you would end up with `https://example.com/preview.php/preview.php/foo.html`

Reply:
To give this more context: Non-deprecated sources for the `$url` always are at least path-absolute, so for them `basePath` vs `baseUrl` does not matter. But legacy or third party insert tags might return a URL that is relative to the `<base href` and so we must use the same base for these URLs as we use in the deprecated `<base href`. This is `Environment::get('base')` which internally uses `->getSchemeAndHttpHost().$request->getBasePath().'/'`

Reply:
I can confirm that using `$request->getUriForPath('/')` for the base path would result in a canonical URL like

```
https://example.com/preview.php/preview.php/de/foobar
``` 

in case the stored `canonicalLink` (directly or indirectly via insert tag) is not at least path absolute.

Reply:
Thanks for the clarification 🙃